### DISCUSSION TITLE
Hướng dẫn sử dụng `ibus-bamboo` trên Wayland

### DISCUSSION THREAD

**Discussion by songoffireandice03** [BambooEngine/ibus-bamboo#521] [Q&A] [answered]
Mình xin nhờ ad (hay người phát triển package này) hỗ trợ giúp mình về việc điều chỉnh để chạy ibus ổn trên Wayland.
Thông tin về máy hiện tại:
OS: Arch Linux
DE: KDE Plasma 6 (Wayland)
Cấu hình .zshrc:
![image](https://github.com/user-attachments/assets/e8d88a19-07e4-4ba0-950d-4932fc943ece)
Cấu hình .zshenv (nếu cần để sửa lỗi)
![image](https://github.com/user-attachments/assets/aeb1d2fa-0d54-4fd2-9f58-c4e09ac60269)
Lỗi gặp: Không đổi ngôn ngữ khi nhấn tổ hợp `Meta (hay nút Windows) + Space` phải đưa chuột qua để chuyển ngôn ngữ trên thanh taskbar rất khó chịu, rê con trỏ qua các ô Bảng mã, Kiểu gõ trên Wayland thì bị bật tắt và gây crash màn hình sau vài lần thử. Ngoài ra thì khi chuyển bàn phím ảo trên KDE thì xuật hiện hộp thông báo như ở ảnh dưới:  
![image](https://github.com/user-attachments/assets/76cc8482-0b73-4d66-b347-9e8f724e9a16)

**Comment by goatastronaut0212** [comment, marked as answer]
@songoffireandice03 

Ibus cảnh báo bạn hãy xóa QT_IM_MODULE và GTK_MODULE ra khỏi biến môi trường nên bạn hãy làm theo những gì nó nói.

Mình không khuyến khích sử dụng Wayland ở thời điểm hiện tại nếu bạn thực sự là 1 người dùng cuối. Ngoài GNOME hoạt động tốt với Ibus thì còn lại bạn cần phải vọc. Cá nhân mình thực sự không có lý do hay thêm động lực nào để thêm hay sửa ibus-bamboo trên Wayland cho tới khi các phần mềm upstream hỗ trợ Wayland tốt hơn như Ibus hay Wayland-protocol. Đây là danh sách các DE/WM hiện có v1 input protocol của Wayland bạn có thể tham khảo tuy vậy mình không dám chắc bạn sẽ làm thế nào cho chúng hoạt động tốt nên là chúc bạn may mắn

|DE/WM|Input protocol v1|
|-|-|
|Gnome|Hỗ trợ tốt nhất|
|KDE| Hỗ trợ|
|Sway|Không hỗ trợ|
|Hyprland|Hỗ trợ|
|Qtile|Hỗ trợ (Không hiển thị icon nếu sử dụng bar)|
|River|Không hỗ trợ|

Nếu muốn mọi thứ trở nên ít tồi tệ hơn. Hãy đóng góp code thêm tại https://github.com/ibus/ibus và https://gitlab.freedesktop.org/wayland/wayland-protocols

Thông tin thêm. Hiện nay từ phiên bản sau trở đi của Ibus đã hỗ trợ wayland với input protocol v2 và có thể hoạt động trên các WM và DE. Các bạn hãy kiểm tra package Ibus trên distro của mình để có thể thử nghiệm và có 1 trải nghiệm tốt hơn
https://github.com/ibus/ibus/releases/tag/1.5.32-rc1
